src/Application.vala

```diff
@@ -230,32 +231,43 @@ public class Terminal.Application : Gtk.Application {
         var window = (MainWindow) active_window;
         bool new_window;
 
-        if (window == null || options.lookup ("new-window", "b", out new_window) && new_window) {
+        if (window == null || (options.lookup ("new-window", "b", out new_window) && new_window)) {
             /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
              * Currently they are set to restore (subject to the restore-tabs setting).
              * If it is desired that tabs should never be restored in these circimstances add another check below.
              */
-            bool restore_tabs = !("commandline" in options || "execute" in options) || window == null;
-            window = new MainWindow (this, restore_tabs);
+            bool restore_tabs = !("commandline" in options || "execute" in options || "working-directory" in options) || window == null;
+            window = new MainWindow (this, restore_tabs, false);
```

Comment:
you can lookup for "new-tab" earlier and then do:
```suggestion
            window = new MainWindow (this, restore_tabs, !new_tab);
```

Reply:
I have removed the third parameter for window construction now